archive/compression/compression.go

```diff
@@ -139,3 +139,13 @@ func (compression *Compression) Extension() string {
 	}
 	return ""
 }
+
+// GetCompression returns the compression method of the reader r which is returned by the method DecompressStream
```

Comment:
nit: break the comment into 2 lines. Also instead of `the reader r` -> `the given reader`, the variable name is not relevant to the caller.